walk/walk.go

```diff
@@ -122,72 +122,72 @@ func Walk(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode,
 		log.Printf("error loading .bazelignore: %v", err)
 	}
 
-	var visit func(*config.Config, string, string, bool)
-	visit = func(c *config.Config, dir, rel string, updateParent bool) {
-		haveError := false
-
-		// TODO: OPT: ReadDir stats all the files, which is slow. We just care about
-		// names and modes, so we should use something like
-		// golang.org/x/tools/internal/fastwalk to speed this up.
-		ents, err := os.ReadDir(dir)
-		if err != nil {
-			log.Print(err)
-			return
-		}
+	visit(c, cexts, isBazelIgnored, knownDirectives, updateRels, wf, c.RepoRoot, "", false)
+}
 
-		f, err := loadBuildFile(c, rel, dir, ents)
-		if err != nil {
-			log.Print(err)
-			if c.Strict {
-				// TODO(https://github.com/bazelbuild/bazel-gazelle/issues/1029):
-				// Refactor to accumulate and propagate errors to main.
-				log.Fatal("Exit as strict mode is on")
-			}
-			haveError = true
-		}
+func visit(c *config.Config, cexts []config.Configurer, isBazelIgnored isIgnoredFunc, knownDirectives map[string]bool, updateRels *UpdateFilter, wf WalkFunc, dir, rel string, updateParent bool) {
```

Comment:
The number of args here might be making it worth wrapping some of these fields in a struct?

Reply:
I actually prefer it this way, structs always make me suspicious of something else modifying it after the initial struct literal.